source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -1,3324 +1,3324 @@
-#endif // _DEBUG
-        if (!mappedAttrBuf[attrId])
-            continue;
-
-        bool rebind = false;
-        if (mappedAttrBuf[attrId]!=attribBufs[attrId])
-        {
-            mappedAttrBuf[attrId]->drop();
-            mappedAttrBuf[attrId] = attribBufs[attrId];
-            mappedAttrBuf[attrId]->grab();
-            rebind = true;
-        }
-        if (attrOffset[attrId]!=offsets[attrId])
-        {
-            attrOffset[attrId] = offsets[attrId];
-            rebind = true;
-        }
-        if (attrStride[attrId]!=strides[attrId])
-        {
-            attrStride[attrId] = strides[attrId];
-            rebind = true;
-        }
-
-        if (rebind||mappedAttrBuf[attrId]->getLastTimeReallocated()>lastValidated)
-            extGlVertexArrayVertexBuffer(vao,attrId,mappedAttrBuf[attrId]->getOpenGLName(),attrOffset[attrId],attrStride[attrId]);
-    }
-
-    bool rebind = false;
-    if (indexBuf!=mappedIndexBuf)
-    {
-        if (indexBuf)
-            indexBuf->grab();
-        if (mappedIndexBuf)
-            mappedIndexBuf->drop();
-        mappedIndexBuf = indexBuf;
-        rebind = true;
-    }
-    else if (mappedIndexBuf&&mappedIndexBuf->getLastTimeReallocated()>lastValidated)
-        rebind = true;
-
-    if (rebind)
-    {
-        if (mappedIndexBuf)
-            extGlVertexArrayElementBuffer(vao,mappedIndexBuf->getOpenGLName());
-        else
-            extGlVertexArrayElementBuffer(vao,0);
-    }
-
-    lastValidated = beginStamp;
-}
-
-bool COpenGLDriver::SAuxContext::setActiveVAO(const COpenGLVAOSpec* const spec, const IGPUMeshBuffer* correctOffsetsForXFormDraw)
-{
-    if (!spec)
-    {
-        CurrentVAO = HashVAOPair(COpenGLVAOSpec::HashAttribs(),nullptr);
-        extGlBindVertexArray(0);
-        freeUpVAOCache(true);
-        return false;
-    }
-
-    const COpenGLVAOSpec::HashAttribs& hashVal = spec->getHash();
-	if (CurrentVAO.first!=hashVal)
-    {
-        auto it = std::lower_bound(VAOMap.begin(),VAOMap.end(),HashVAOPair(hashVal,nullptr),[](HashVAOPair lhs, HashVAOPair rhs) -> bool { return lhs.first < rhs.first; });
-        if (it != VAOMap.end() && it->first==hashVal)
-            CurrentVAO = *it;
-        else
-        {
-            COpenGLVAO* vao = new COpenGLVAO(spec);
-            CurrentVAO = HashVAOPair(hashVal,vao);
-            VAOMap.insert(it,CurrentVAO);
-        }
-
-        #ifdef _DEBUG
-            assert(!(CurrentVAO.second->getDebugHash()!=hashVal));
-        #endif // _DEBUG
-
-        extGlBindVertexArray(CurrentVAO.second->getOpenGLName());
-    }
-
-    if (correctOffsetsForXFormDraw)
-    {
-        size_t offsets[asset::EVAI_COUNT] = {0};
-        memcpy(offsets,&spec->getMappedBufferOffset(asset::EVAI_ATTR0),sizeof(offsets));
-        for (size_t i=0; i<asset::EVAI_COUNT; i++)
-        {
-            if (!spec->getMappedBuffer((asset::E_VERTEX_ATTRIBUTE_ID)i))
-                continue;
-
-            if (spec->getAttribDivi
```

Comment:
I said to assume `Params.HandleSRGB=true` and remove it from the engine!

Reply:
allright